_layouts/default.html

```diff
@@ -2,7 +2,7 @@
 <html lang="en-us">
   <head>
     <meta charset="UTF-8">
-    <title>{{ page.title | default: site.title }}</title>
+    <title>{{ site.title | default: site.github.repository_name }} by {{ site.github.owner_name }}</title>
```

Comment:
I agree with @sethjackson.
The title should take value from `site.title` **if it exists**, else fallback to `page.title` which will either be set in the Front Matter or generated by the `jekyll-titles-from-headings` plugin.
so it should ideally be:
```
<title>{{ site.title | default: page.title }}</title>
```
because if its 
```
<title>{{ page.title | default: site.title }}</title>
```
then, `page.title` which is **never `nil`** will **always override** `default: site.title`